dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/ReadTimeoutTest.java

```diff
@@ -0,0 +1,124 @@
+
+    private static ServerConfiguration emptyConfiguration = null;
+
+    /**
+     * Updates the sample, which is expected to be at the hard-coded path.
+     * If this test is failing, check this path is correct.
+     */
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Add LDAP variables to bootstrap properties file
+        LDAPUtils.addLDAPVariables(server);
+        Log.info(c, "setUp", "Starting the server... (will wait for userRegistry servlet to start)");
+        server.copyFileToLibertyInstallRoot("lib/features", "internalfeatures/securitylibertyinternals-1.0.mf");
+        server.addInstalledAppForValidation("userRegistry");
+        server.startServer(c.getName() + ".log");
+
+        //Make sure the application has come up before proceeding
+        assertNotNull("Application userRegistry does not appear to have started.",
+                      server.waitForStringInLog("CWWKZ0001I:.*userRegistry"));
+        assertNotNull("Security service did not report it was ready",
+                      server.waitForStringInLog("CWWKS0008I"));
+        assertNotNull("Server did not came up",
+                      server.waitForStringInLog("CWWKF0011I"));
+
+        Log.info(c, "setUp", "Creating servlet connection the server");
+        servlet = new UserRegistryServletConnection(server.getHostname(), server.getHttpDefaultPort());
+
+        servlet.getRealm();
+        /*
+         * The original server configuration has no registry or Federated Repository configuration.
+         */
+        emptyConfiguration = server.getServerConfiguration();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        Log.info(c, "tearDown", "Stopping the server...");
+        try {
+            server.stopServer("CWWKG0033W", "CWWKG0014E");
+        } finally {
+            server.deleteFileFromLibertyInstallRoot("lib/features/internalfeatures/securitylibertyinternals-1.0.mf");
+        }
+    }
+
+//    @Test
+//    public void testReadTimeout() throws Exception { // for issue
+//        Log.info(c, "testReadTimeout", "Entering test testReadTimeout");
+//
+//        // Checking that these are correctly logged in the trace
+//        String tr = "iReadTimeout=60000";
+//        List<String> errMsgs = server.findStringsInLogsAndTrace(tr);
+//        assertFalse("Should have found, " + tr, errMsgs.isEmpty());
+//    }
+
+    @Test
+    public void testCustomReadTimeout() throws Exception {
+        Log.info(c, "testCustomContextAndCache", "Entering test testCustomContextAndCache");
+
+        ServerConfiguration clone = emptyConfiguration.clone();
+        LdapRegistry ldap = clone.getLdapRegistries().get(0);
+//        LdapRegistry ldap = createTDSLdapRegistry(clone, "LDAP", "SampleLdapIDSRealm");
+//        createFederatedRepository(clone, "OneLDAPRealm", new String[] { ldap.getBaseDN() });
+
+        String tr = ", iReadTimeout=60000,"; // TODO Might be null?
```

Comment:
Rename this field. We have classes that you use statically named Tr, so it is confusing seeing this named "tr".